torch/distributed/distributed_c10d.py

```diff
@@ -2656,7 +2655,7 @@ def peer_kwarg(op: P2POp) -> Dict[str, int]:
         key = "group_dst" if op.op == isend else "group_src"
         return {key: op.group_peer}
 
-    if device.type == "cuda":
+    if device.type in ["cuda", "xpu"]:
```

Comment:
This change may have conflict with a PR we try to land currently. 
https://github.com/pytorch/pytorch/pull/135338/files#diff-61109d1cb2a0bd13fc51d678a82666295289da1ec0a1a694e73d9e8c28f51bdcL2659-R2670
Would you mind rebasing this PR once the other PR has landed? Thanks!
cc: @taozhiwei @c-p-i-o

Reply:
No problem. Then I would like to revert this change in my PR and add `supports_coalescing` in XPU distributed backend by following https://github.com/pytorch/pytorch/pull/135338.